CI: macOS conda-forge job failing because of scikit-umfpack upgrade

`scikit-umfpack` 0.4.2 was released in July, the first release in 1.5 years. The only place we test this optional dependency is in the macOS conda-forge based CI job (because we can only get binaries from conda-forge). The upgrade in conda-forge landed yesterday: https://github.com/conda-forge/scikit-umfpack-feedstock/pull/43. That change seems to have caused a few failures because of warnings ([example CI log](https://github.com/scipy/scipy/actions/runs/17472284076/job/49679084840?pr=23557)):

```

__________________ TestFactorized.test_singular_with_umfpack ___________________
[gw4] darwin -- Python 3.11.13 /Users/runner/miniconda3/envs/scipy-dev/bin/python3.11

self = <scipy.sparse.linalg._dsolve.tests.test_linsolve.TestFactorized object at 0x138363e50>

    @pytest.mark.skipif(not has_umfpack, reason="umfpack not available")
    def test_singular_with_umfpack(self):
        use_solver(useUmfpack=True)
        with warnings.catch_warnings():
            msg = "divide by zero encountered in double_scalars"
            warnings.filterwarnings("ignore", msg, RuntimeWarning)
>           assert_warns(umfpack.UmfpackWarning, self._check_singular)
E           DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

msg        = 'divide by zero encountered in double_scalars'
self       = <scipy.sparse.linalg._dsolve.tests.test_linsolve.TestFactorized object at 0x138363e50>

scipy/sparse/linalg/_dsolve/tests/test_linsolve.py:87: DeprecationWarning

...


>           warn('spsolve requires A be CSC or CSR matrix format',
                 SparseEfficiencyWarning, stacklevel=2)
E           scipy.sparse._base.SparseEfficiencyWarning: spsolve requires A be CSC or CSR matrix format

A          = <Compressed Sparse Column sparse array of dtype 'float64'
	with 2 stored elements and shape (2, 2)>
b          = [1, 6]
is_pydata_sparse = False
permc_spec = None
pydata_sparse_cls = None
use_umfpack = True

scipy/sparse/linalg/_dsolve/linsolve.py:223: SparseEfficiencyWarning
_ build-install/usr/lib/python3.11/site-packages/scipy/interpolate/tests/test_bsplines.py _
[gw3] darwin -- Python 3.11.13 /Users/runner/miniconda3/envs/scipy-dev/bin/python3.11
worker 'gw3' crashed while running 'build-install/usr/lib/python3.11/site-packages/scipy/interpolate/tests/test_bsplines.py::TestMakeND::test_2D_mixed[k1]'
_ build-install/usr/lib/python3.11/site-packages/scipy/interpolate/tests/test_bsplines.py _
[gw5] darwin -- Python 3.11.13 /Users/runner/miniconda3/envs/scipy-dev/bin/python3.11
worker 'gw5' crashed while running 'build-install/usr/lib/python3.11/site-packages/scipy/interpolate/tests/test_bsplines.py::TestMakeND::test_2D_mixed[k2]'

...

                    elif sym_pos:
                        sym_pos = False
>                       warn(
                            "Solving system with option 'sym_pos':True "
                            "failed. It is normal for this to happen "
                            "occasionally, especially as the solution is "
                            "approached. However, if you see this frequently, "
                            "consider setting option 'sym_pos' to False.",
E                           scipy.optimize._optimize.OptimizeWarning: Solving system with option 'sym_pos':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'sym_pos' to False.
```

The `Solving system with option 'sym_pos':True failed` warning happens a number of times. Crashes are visible in the log:

```
........................................................................ [ 79%]
Fatal Python error: Segmentation fault

..........[gw5] node down: Not properly terminated
Thread 0x000000016c4cb000 (most recent call first):
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/execnet/gateway_base.py", line 534 in read
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/execnet/gateway_base.py", line 567 in from_io
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/execnet/gateway_base.py", line 1160 in _thread_receiver
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/execnet/gateway_base.py", line 341 in run
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/execnet/gateway_base.py", line 411 in _perform_spawn

F
Current thread 0x00000002086e5f00 (most recent call first):
replacing crashed worker gw5
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/scikits/umfpack/_umfpack.py", line 426 in umfpack_dl_symbolic
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/scikits/umfpack/umfpack.py", line 524 in symbolic
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/scikits/umfpack/umfpack.py", line 557 in numeric
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/scikits/umfpack/umfpack.py", line 772 in linsolve
  File "/Users/runner/work/scipy/scipy/build-install/usr/lib/python3.11/site-packages/scipy/sparse/linalg/_dsolve/linsolve.py", line 270 in spsolve
  File "/Users/runner/work/scipy/scipy/build-install/usr/lib/python3.11/site-packages/scipy/interpolate/_ndbspline.py", line 489 in make_ndbspl
  File "/Users/runner/work/scipy/scipy/build-install/usr/lib/python3.11/site-packages/scipy/interpolate/tests/test_bsplines.py", line 2940 in test_2D_mixed
  File "/Users/runner/miniconda3/envs/scipy-dev/lib/python3.11/site-packages/_pytest/python.py", line 157 in pytest_pyfunc_call
```

```
=========================== short test summary info ============================
FAILED scipy/interpolate/tests/test_bsplines.py::TestMakeND::test_2D_mixed[k0]
FAILED scipy/optimize/_trustregion_constr/tests/test_projections.py::TestProjections::test_compare_dense_and_sparse
FAILED scipy/sparse/tests/test_array_api.py::test_inv[csc]
FAILED scipy/sparse/linalg/_dsolve/tests/test_linsolve.py::TestFactorized::test_singular_with_umfpack - DeprecationWarning: builtin type SwigPyObject has no __module__ attribute
FAILED scipy/sparse/linalg/_dsolve/tests/test_linsolve.py::TestLinsolve::test_shape_compatibility - scipy.sparse._base.SparseEfficiencyWarning: spsolve requires A be CSC or CSR matrix format
FAILED scipy/interpolate/tests/test_bsplines.py::TestMakeND::test_2D_mixed[k1]
FAILED scipy/interpolate/tests/test_bsplines.py::TestMakeND::test_2D_mixed[k2]
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_unbounded_below_no_presolve_corrected - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'sym_pos':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'sym_pos' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseUmfpack::test_unbounded_below_no_presolve_corrected - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'sym_pos':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'sym_pos' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparse::test_unbounded_below_no_presolve_corrected - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'sym_pos':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'sym_pos' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_unbounded_below_no_presolve_corrected - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'sym_pos':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'sym_pos' to False.
= 11 failed,
```

--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -111,7 +111,8 @@ jobs:
         mamba install python=${{ matrix.python-version}}
 
         # optional test dependencies
-        mamba install scikit-umfpack scikit-sparse
+        # scikit-umfpack is pinned because 0.4.2 packages seem broken, see gh-23564
+        mamba install "scikit-umfpack=0.3.3" scikit-sparse
 
         # configure system boost
         mamba install libboost-headers=1.89.0

